src/main/java/com/datasonnet/plugins/JavaFormatPlugin.java

```diff
@@ -0,0 +1,96 @@
+package com.datasonnet.plugins;
+
+import com.datasonnet.document.Document;
+import com.datasonnet.document.JavaObjectDocument;
+import com.datasonnet.document.StringDocument;
+import com.datasonnet.spi.DataFormatPlugin;
+import com.datasonnet.spi.PluginException;
+import com.datasonnet.spi.UjsonUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import ujson.Value;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JavaFormatPlugin implements DataFormatPlugin {
+    public static String OUTPUT_CLASS = "OutputClass";
+    public static String DATE_FORMAT = "DateFormat";
+
+    public static String DEFAULT_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+    public JavaFormatPlugin() { }
+
+    public Value read(Object input, Map<String, Object> params) throws PluginException {
+        ObjectMapper mapper = new ObjectMapper();
+        DateFormat df = new SimpleDateFormat(params.containsKey(DATE_FORMAT) ? (String)params.get(DATE_FORMAT) : DEFAULT_DATE_FORMAT);
+        mapper.setDateFormat(df);
+        try {
+            String jsonStr = mapper.writeValueAsString(input);
+            return UjsonUtil.jsonObjectValueOf(jsonStr);
+        } catch (JsonProcessingException e) {
+            throw new PluginException(e);
+        }
+    }
+
+    public Document write(Value input, Map<String, Object> params, String mimeType) throws PluginException {
+        try {
+            String jsonString = UjsonUtil.jsonObjectValueTo(input);
+            ObjectMapper mapper = new ObjectMapper();
+            DateFormat df = new SimpleDateFormat(params.containsKey(DATE_FORMAT) ? (String)params.get(DATE_FORMAT) : DEFAULT_DATE_FORMAT);
+            mapper.setDateFormat(df);
+            JsonNode node = mapper.readTree(jsonString);
+
+            JavaType valueType = null;
+
+            if (node.isObject()) {
+                if (params != null && params.containsKey(OUTPUT_CLASS)) {
+                    valueType = mapper.getTypeFactory().constructFromCanonical(params.get(OUTPUT_CLASS).toString());
+                } else {
+                    valueType = mapper.getTypeFactory().constructRawMapType(HashMap.class);
+                }
+            } else if (node.isArray()) {
```

Comment:
I'm not sure we want to get so magic here based on what sort of node we're handed. For example, this makes it impossible for someone to specify they should get a List of a particular sort of bean (which is possible using constructFromCanonical). Less commonly, there are also alternative sequences that someone could use that jackson can load into.

Reply:
Yes, I'm already thinking about a way to provide a different/custom deserializer to handle these types of scenarios, and a parameter(s) to specify array of beans vs list of beans vs list of maps.

Reply:
Already supported by constructFromCanonical, even! For that you just use java generics syntax: `java.util.List<com.example.MyBean>`. There's even fancier stuff possible too, beyond what generics support... the documentation is spotty, though

Reply:
OK, I think here's what we should do - if the output class is set explicitly, it calls constructFromCanonical, otherwise tries to resolve based on the type of the node.

Reply:
Yeah, I think I like that